### PR TITLE
Feature/sim card network preference

### PR DESCRIPTION
**Pull request by rajjy1**

### CODE REVIEW

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        public override string PostPath => "network_preferences";
+
+        public MobileOperatorNetworksPreferencesRecord Get(string id, BaseOptions baseOptions, RequestOptions requestOptions = null, string postFix = "network_preferences")
+        {
+            return this.GetEntity(id, baseOptions, requestOptions, postFix);
+        }
+
+        public async Task<MobileOperatorNetworksPreferencesRecord> GetAsync(string id, BaseOptions baseOptions, RequestOptions requestOptions = null, CancellationToken cancellationToken = default, string postFix = "network_preferences")
+        {
+            return await this.GetEntityAsync(id, baseOptions, requestOptions, cancellationToken, postFix);
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs:43]: Pass this.PostPath instead and remove the postFix param?

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        /// <param name="apiKey">api key.</param>
+        internal SimCardNetworkPreferenceService(string apiKey)
+            : base(apiKey)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override string BasePath => "/sim_cards";
+
+        public override string PostPath => "network_preferences";
```
[the remark this answers is not in the document]
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs:34]: It is used in delete and update. Also, it is not passed in the parameter in the method. It is passed internally.

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs (hunk 1 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardNetworkPreferenceService.cs:43]: Done.

**Comment by rajjy1** [comment]
> Looks good, I just think we should remove the option to past postPath urls and just use the one inside the Service classes

That is a good idea. Done.

**Pull request merged**